nav2_bringup/src/loc_and_nav.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021, Samsung Research America
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include <memory>
+#include <vector>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_map_server/map_server.hpp"
+#include "nav2_amcl/amcl_node.hpp"
+#include "nav2_controller/nav2_controller.hpp"
+#include "nav2_planner/planner_server.hpp"
+#include "nav2_recoveries/recovery_server.hpp"
+#include "nav2_lifecycle_manager/lifecycle_manager.hpp"
+#include "nav2_bt_navigator/bt_navigator.hpp"
+#include "nav2_waypoint_follower/waypoint_follower.hpp"
+
+template<typename NodeT>
+std::shared_ptr<std::thread> create_spin_thread(NodeT & node)
+{
+  return std::make_shared<std::thread>(
+    [node]() {
+      rclcpp::executors::SingleThreadedExecutor executor;
+      executor.add_node(node->get_node_base_interface());
+      executor.spin();
+      executor.remove_node(node->get_node_base_interface());
+    });
+}
+
+int main(int argc, char ** argv)
```

Comment:
OK, So I want a biiiiiiiig block of comments before this main function. At the very least:
- Explaining why single threaded executors and not multithreaded (higher CPU)
- Explaining why not Intraprocess comms (higher CPU if not all using, and no support for QOS)
- Links to that discourse thread / any related tickets 
- Explaining that this is an example for the default nav2 servers and bring up in localization mode. It is our expectation for an application specific thing you're mirroring `nav2_bringup` package and modifying it for your sp. maps/robots/bringup needs so this is an applied and working demonstration for the default system bringup ONLY

Reply:
This is all tribal knowledge that should remain close to the code for future generations